### PR TITLE
chore: update tables for human annotations on experiment runs

### LINKED ISSUE #3425: [datasets] adjust tables to allow for human evaluations
As a user, I'd like to add human grading or comments via explanations on the experiment runs

### PR DESCRIPTION
**Pull request by axiomofjoy**
- Renames `experiment_evaluations` table to `experiment_annotations`.
- Adds `annotator_kind` column with a check constraint.
- Updates source and tests.

resolves #3425

### CODE REVIEW

File: src/phoenix/db/migrations/versions/10460e46d750_datasets.py [hunk 1]
```diff
@@ -216,6 +216,15 @@ def upgrade() -> None:
             sa.String,
             nullable=False,
         ),
+        sa.Column(
+            "annotator_kind",
+            sa.String,
+            sa.CheckConstraint(
+                "annotator_kind IN ('LLM', 'HUMAN')",
```
Comment (mikeldking) [inline src/phoenix/db/migrations/versions/10460e46d750_datasets.py:223]:
```suggestion
                "annotator_kind IN ('LLM', 'CODE', 'HUMAN')",
```
We are going to have pure code based evaluations so we should probably leave room for that

File: src/phoenix/db/models.py [hunk 2]
```diff
@@ -538,14 +538,17 @@ class ExperimentRun(Base):
...
+class ExperimentAnnotation(Base):
+    __tablename__ = "experiment_annotations"
     id: Mapped[int] = mapped_column(primary_key=True)
     experiment_run_id: Mapped[int] = mapped_column(
         ForeignKey("experiment_runs.id", ondelete="CASCADE"),
         index=True,
     )
     name: Mapped[str]
+    annotator_kind: Mapped[str] = mapped_column(
+        CheckConstraint("annotator_kind IN ('LLM', 'HUMAN')", name="valid_annotator_kind"),
```
Comment (mikeldking) [inline src/phoenix/db/models.py:550]:
```suggestion
        CheckConstraint("annotator_kind IN ('LLM', 'CODE', 'HUMAN')", name="valid_annotator_kind"),
```

File: src/phoenix/db/migrations/versions/10460e46d750_datasets.py (hunk 1 above)
Comment (axiomofjoy) [reply src/phoenix/db/migrations/versions/10460e46d750_datasets.py]: addressed in de0df5f60ab4b38672f2e289c599b1f24e8851da

File: src/phoenix/db/models.py (hunk 2 above)
Comment (axiomofjoy) [reply src/phoenix/db/models.py]: addressed in de0df5f60ab4b38672f2e289c599b1f24e8851da

**Review by mikeldking — APPROVED** [review]

**Pull request merged**